dali/python/nvidia/dali/external_source.py

```diff
@@ -251,10 +277,14 @@ class ExternalSource():
     For the GPU, to avoid extra copy, the provided buffer must be contiguous. If you provide a list
     of separate Tensors, there will be an additional copy made internally, consuming both memory
     and bandwidth.
+
+`batch` : optional
+    If set to ``True`` or ``None``, the ``source`` is expected to produce an entire batch at once.
+    If set to ``False``, the ``source`` is called per-sample. Its first parameter is sample index.
 """
 
     def __init__(self, source = None, num_outputs = None, *, cycle = None, layout = None, name = None, device = "cpu",
-                 cuda_stream = None, use_copy_kernel = None, **kwargs):
+                 cuda_stream = None, use_copy_kernel = None, batch = None, **kwargs):
```

Comment:
How about calling it `batch_mode`? Seems like a better description of what it does

Reply:
I think we already boolean argument called `batch`.

Reply:
It is for normalized, but I don't know if we can call it a corresponding example.

Reply:
In any case, the documentation is rather clear. I'll change `batch : optional` to  `batch : bool, optional`.